packages/atomic/src/components/common/interface/interface-common.tsx

```diff
@@ -2,18 +2,158 @@ import type {SearchEngine} from '@coveo/headless';
 import {LogLevel} from '@coveo/headless';
 import type {RecommendationEngine} from '@coveo/headless/recommendation';
 import type {InsightEngine} from '@coveo/headless/insight';
-import {i18n} from 'i18next';
+import {i18n, TFunction} from 'i18next';
+import {ComponentInterface, h} from '@stencil/core';
+import {setCoveoGlobal} from '../../../global/environment';
+import {loadFocusVisiblePolyfill} from '../../../global/focus-visible';
+import {initi18n} from './i18n';
+import {HTMLStencilElement} from '@stencil/core/internal';
+import {loadDayjsLocale} from '../../../utils/dayjs-locales';
+import {InitializeEvent} from '../../../utils/initialization-utils';
+import {AnyBindings} from './bindings';
+import {
+  i18nBackendOptions,
+  i18nTranslationNamespace,
+} from '../../common/interface/i18n';
+import Backend from 'i18next-http-backend';
 
 export type AnyEngineType = SearchEngine | RecommendationEngine | InsightEngine;
 
-export interface BaseAtomicInterface<EngineType extends AnyEngineType> {
+export interface BaseAtomicInterface<EngineType extends AnyEngineType>
+  extends ComponentInterface {
   analytics: boolean;
   i18n: i18n;
   engine?: EngineType;
   languageAssetsPath: string;
   iconAssetsPath: string;
   logLevel?: LogLevel;
   language: string;
+  host: HTMLStencilElement;
+  bindings: AnyBindings;
+  error?: Error;
 
   updateIconAssetsPath(): void;
 }
+
+export class CommonAtomicInterfaceHelper<Engine extends AnyEngineType> {
+  private i18nPromise!: Promise<TFunction>;
+  private hangingComponentsInitialization: InitializeEvent[] = [];
+
+  constructor(
+    private atomicInterface: BaseAtomicInterface<Engine>,
+    globalVariableName: string
+  ) {
+    setCoveoGlobal(globalVariableName);
+    loadFocusVisiblePolyfill();
+
+    const {
+      connectedCallback: originalConnectedCallback,
+      render: originalRender,
+    } = atomicInterface;
+
+    atomicInterface.connectedCallback = () => {
+      this.i18nPromise = initi18n(atomicInterface);
+
+      return (
+        originalConnectedCallback &&
+        originalConnectedCallback.call(atomicInterface)
+      );
+    };
+
+    atomicInterface.render = () => {
+      if (atomicInterface.error) {
+        return (
+          <atomic-component-error
+            element={atomicInterface.host}
+            error={atomicInterface.error}
+          ></atomic-component-error>
+        );
+      }
+
+      return originalRender && originalRender.call(atomicInterface);
+    };
+  }
+
+  public onComponentInitializing(event: InitializeEvent) {
+    event.preventDefault();
+    event.stopPropagation();
+
+    if (this.atomicInterface.engine) {
```

Comment:
just to confirm, is this right? we're checking that the engine exists and resolving with the bindings?

Reply:
Yeah. To be frank, I've just copied what was already done in the search interface in `handleInitialization`.